### DISCUSSION TITLE
Troubleshooting on windows 11

### DISCUSSION THREAD

**Discussion by Br3akp01nt** [mrcjkb/haskell-tools.nvim#171] [Q&A] [answered]
Hey, I've been trying to get this to work on windows 11 but seem unable to do so.
Healthchecks pass fine, but upon opening a haskell file, nvim just goes blank.
Log files only report logging started, even when setting loglevel to debug.
It seems to occur only when ht.setup or ht.start_or_attack is run, even when they are empty.

I've tried running the minimal install, which works, but since setup or start_or_attack isn't run, I can't glean any useful info from that.

Any tips on continued troubleshooting?

**Comment by mrcjkb** [comment]
Hey. Thanks for opening this :smile: 

I'm afraid I don't have access to a Windows device, so I'd have to rely on hopefully getting the logs working to help troubleshoot this.
And if we do get it working, I won't be able to officially support Windows, unfortunately.

* Could you please paste your config here?

* Which method are you trying? The `setup` method or the `ftplugin`/`start_or_attach` one?

* Are you setting the log level to debug in the config or manually?

I do have some places where I make assumptions about the file path separator, concatenating file paths with `/`. 
AFAIK, Windows should support both `/` and `\` separators. But maybe that's causing issues.
Another potential issue could be that the `project_util` module is not able to detect the Windows root directory, causing infinite recursion.

I'll create a branch with some additional logs when I have some spare time, if you're willing to help me debug this :smile:

**Comment by Br3akp01nt** [comment]
Oh, I didn't realize windows was unsupported. I guess it's not very surprising in that case.

I'm still happy to help in any way I can, so for starters I'll post what logs and configs I have:

```
-- ~\AppData\Local\nvim\init.lua

require('plugins')
require('haskell')
```

```
-- ~\AppData\Local\nvim\lua\plugins.lua

local vim = vim
local execute = vim.api.nvim_command
local fn = vim.fn
-- ensure that packer is installed
local install_path = fn.stdpath('data')..'/site/pack/packer/opt/packer.nvim'
if fn.empty(fn.glob(install_path)) > 0 then
    execute('!git clone https://github.com/wbthomason/packer.nvim '..install_path)
    execute 'packadd packer.nvim'
end
vim.cmd('packadd packer.nvim')
local packer = require'packer'
local util = require'packer.util'
packer.init({
  package_root = util.join_paths(vim.fn.stdpath('data'), 'site', 'pack')
})

packer.startup(function(use)
  use('wbthomason/packer.nvim')
  
  use {
    'MrcJkb/haskell-tools.nvim',
    requires = {
      'nvim-lua/plenary.nvim'
    },
    config = function()
    end,
  }

  end
)
```

```
-- ~\AppData\Local\nvim\lua\haskell.lua

local ht = require('haskell-tools')
local buffer = vim.api.nvim_get_current_buf()
local def_opts = { noremap = true, silent = true, }
ht.setup {
  hls = {
    on_attach = function(client, bufnr)
      local opts = vim.tbl_extend('keep', def_opts, { buffer = bufnr, })
      -- haskell-language-server relies heavily on codeLenses,
      -- so auto-refresh (see advanced configuration) is enabled by default
      vim.keymap.set('n', '<space>ca', vim.lsp.codelens.run, opts)
      vim.keymap.set('n', '<space>hs', ht.hoogle.hoogle_signature, opts)
      vim.keymap.set('n', '<space>ea', ht.lsp.buf_eval_all, opts)
    end,
  },
}

-- Suggested keymaps that do not depend on haskell-language-server:
local bufnr = vim.api.nvim_get_current_buf()
-- set buffer = bufnr in ftplugin/haskell.lua
local opts = { noremap = true, silent = true, buffer = bufnr }

-- Toggle a GHCi repl for the current package
vim.keymap.set('n', '<leader>rr', ht.repl.toggle, opts)
-- Toggle a GHCi repl for the current buffer
vim.keymap.set('n', '<leader>rf', function()
  ht.repl.toggle(vim.api.nvim_buf_get_name(0))
end, def_opts)
vim.keymap.set('n', '<leader>rq', ht.repl.quit, opts)
```

As for logging, I tried `ht.log.set_level(vim.log.levels.DEBUG)` but received nil error on vim.log.
Running it manually in nvim `:lua require('haskell-tools').log.set_level(vim.log.levels.DEBUG)` seems to have run successfully, but logs were still only as follows: 
```
~\AppData\Local\nvim-data\haskell-tools.log

[START][2023-04-06 19:47:51] haskell-tools.nvim logging initiated
[START][2023-04-06 19:48:03] haskell-tools.nvim logging initiated
[START][2023-04-06 19:48:19] haskell-tools.nvim logging initiated
[START][2023-04-06 19:50:40] haskell-tools.nvim logging initiated
[START][2023-04-06 19:50:58] haskell-tools.nvim logging initiated
[START][2023-04-06 19:51:01] haskell-tools.nvim logging initiated
[START][2023-04-06 19:51:04] haskell-tools.nvim logging initiated
[START][2023-04-06 19:57:45] haskell-tools.nvim logging initiated
```

The corresponding hls log file was empty.


Note: I have also tried using haskell.lua from `~\AppData\Local\nvim\ftplugin` as well as calling the setup using both `setup` and `start_or_attach`


Thank you for responding and let me know if you need more info, I'm happy to help.

**Reply by mrcjkb** [reply, marked as answer]
Thanks for the input. I've moved this to #172 for now :smile:
